plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -651,11 +653,30 @@ int vtkF3DOCCTReader::RequestData(
     STEPCAFControl_Reader reader;
     TransferToDocument(this, reader, doc);
   }
-  else // FILE_FORMAT::IGES
+  else if (this->FileFormat == FILE_FORMAT::IGES)
   {
     IGESCAFControl_Reader reader;
     TransferToDocument(this, reader, doc);
   }
+  else if (this->FileFormat == FILE_FORMAT::XBF)
+  {
+    try
+    {
+      Handle(TDocStd_Application) app = new TDocStd_Application();
+      BinXCAFDrivers::DefineFormat(app);
+      if (app->Open(this->GetFileName().c_str(), doc) != PCDM_RS_OK)
+      {
+        vtkErrorWithObjectMacro(this, "Failed to read XBF file");
+        return 1;
+      }
+    }
+    catch (Standard_Failure const& failure)
+    {
+      vtkErrorWithObjectMacro(
+        this, "Could not initialize XCAF reader " << failure.GetMessageString());
```

Comment:
is this reachable with an invalid file ?

Reply:
This one I'm not sure, it looks like it would be a initialization issue before we even touch the file

Reply:
of occt  ?

Reply:
Yeah I'm not sure if/when it can happen actually, maybe I should just remove this `try`/`catch`?

Reply:
yeah thats fine imo.